src/main/java/com/prgms/VoucherApp/domain/VoucherCreator.java

```diff
@@ -0,0 +1,31 @@
+package com.prgms.VoucherApp.domain;
+
+import com.prgms.VoucherApp.storage.VoucherStorage;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class VoucherCreator {
+
+    private final VoucherStorage voucherStorage;
+
+    public VoucherCreator(VoucherStorage voucherStorage) {
+        this.voucherStorage = voucherStorage;
+    }
+
+    public Voucher createVoucher(VoucherPolicy voucherPolicy, long amount) {
+        Voucher voucher = null;
+        switch (voucherPolicy) {
+            case FIXED_VOUCHER:
+                voucher = new FixedAmountVoucher(UUID.randomUUID(), amount);
+                break;
+            case PERCENT_VOUCHER:
+                voucher = new PercentDiscountVoucher(UUID.randomUUID(), amount);
+        }
```

Comment:
```suggestion
        var voucher = switch (voucherPolicy) {
            case FIXED_VOUCHER -> new FixedAmountVoucher(UUID.randomUUID(), amount);

            case PERCENT_VOUCHER ->  new PercentDiscountVoucher(UUID.randomUUID(), amount);
        }
```
향상된 스위치문을 사용하시면 불필요한 변수할당을 제거할 수 있겠네요